tests/testthat/test-tfrmt_sigdig.R

```diff
@@ -211,3 +211,111 @@ test_that("tfrmt_sigdig returns a tfrmt", {
 
 })
 
+
+test_that("varying group/label inputs",{
+
+  dat_sigdig <- tibble::tribble(
+    ~group1,  ~ group2, ~ lbl, ~ sigdig,
+    "CHEM", "BILIRUBIN", "v1", 1,
+    "CHEM",  ".default", "v2",1,
+    "HEM", "CHOLESTEROL", "v1",2,
+    ".default", "EOSINOPHILS", "v1",2
+  )
+
+  # if no group or label, assume all non-sigdig columns are groups
+  t_out <- tfrmt_sigdig(dat_sigdig)
+
+  expect_equal(t_out$group, vars(group1, group2, lbl), ignore_attr = TRUE)
+
+  expect_equal(t_out$label, quo())
+
+  # if only some are specified, assume the rest are groups
+  t_out <- tfrmt_sigdig(dat_sigdig, label = group2)
+
+  expect_equal(t_out$group, vars(group1, lbl), ignore_attr = TRUE)
+  expect_equal(t_out$label, quo(group2), ignore_attr = TRUE)
+
+  # including a group that is not present in the data
+  t_out <- tfrmt_sigdig(dat_sigdig, group = vars(lbl, grp3), label = group2)
+
+  expect_equal(t_out$group, vars(lbl, grp3, group1), ignore_attr = TRUE)
```

Comment:
So I am not sure if this would be my expected functionality. I think I would assume if I provide groups it will only use what I provide as groups. I think I would want it to at least warning here saying grp3 isn't in signing. But I would also not have group1 as a group at all . Basically the supplied grouping should supersede the provided data

Reply:
Okay, that makes sense. I think that is clearer  - either you provide groups and it will use that OR it will use the data, but not both. I can update.